lighthouse-core/lib/task-groups.js

```diff
@@ -5,90 +5,108 @@
-  'Invalidate Layout': groupIdToName.styleLayout,
-  'JS Frame': groupIdToName.scripting,
-  'Input Latency': groupIdToName.scripting,
-  'Layout': groupIdToName.styleLayout,
-  'Major GC': groupIdToName.scriptGC,
-  'DOMContentLoaded event': groupIdToName.scripting,
-  'First paint': groupIdToName.painting,
-  'FMP': groupIdToName.painting,
-  'FMP candidate': groupIdToName.painting,
-  'Load event': groupIdToName.scripting,
-  'Minor GC': groupIdToName.scriptGC,
-  'Paint': groupIdToName.painting,
-  'Paint Image': groupIdToName.images,
-  'Paint Setup': groupIdToName.painting,
-  'Parse Stylesheet': groupIdToName.parseHTML,
-  'Parse HTML': groupIdToName.parseHTML,
-  'Parse Script': groupIdToName.scriptParseCompile,
-  'Other': groupIdToName.other,
-  'Rasterize Paint': groupIdToName.painting,
-  'Recalculate Style': groupIdToName.styleLayout,
-  'Request Animation Frame': groupIdToName.scripting,
-  'Request Idle Callback': groupIdToName.scripting,
-  'Request Main Thread Frame': groupIdToName.painting,
-  'Image Resize': groupIdToName.images,
-  'Finish Loading': groupIdToName.loading,
-  'Receive Data': groupIdToName.loading,
-  'Receive Response': groupIdToName.loading,
-  'Send Request': groupIdToName.loading,
-  'Run Microtasks': groupIdToName.scripting,
-  'Schedule Style Recalculation': groupIdToName.styleLayout,
-  'Scroll': groupIdToName.compositing,
-  'Task': groupIdToName.other,
-  'Timer Fired': groupIdToName.scripting,
-  'Install Timer': groupIdToName.scripting,
-  'Remove Timer': groupIdToName.scripting,
-  'Timestamp': groupIdToName.scripting,
-  'Update Layer': groupIdToName.compositing,
-  'Update Layer Tree': groupIdToName.compositing,
-  'User Timing': groupIdToName.scripting,
-  'Create WebSocket': groupIdToName.scripting,
-  'Destroy WebSocket': groupIdToName.scripting,
-  'Receive WebSocket Handshake': groupIdToName.scripting,
-  'Send WebSocket Handshake': groupIdToName.scripting,
-  'XHR Load': groupIdToName.scripting,
-  'XHR Ready State Change': groupIdToName.scripting,
+/**
+ * Make sure the traceEventNames keep up with the ones in DevTools
+ * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/timeline_model/TimelineModel.js?type=cs&q=TimelineModel.TimelineModel.RecordType+%3D&g=0&l=1156
+ * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/timeline/TimelineUIUtils.js?type=cs&q=_initEventStyles+-f:out+f:devtools&sq=package:chromium&g=0&l=39
+ */
+const taskGroups = {
+  parseHTML: {
+    id: '',
+    label: 'Parse HTML & CSS',
+    traceEventNames: ['ParseHTML', 'ParseAuthorStyleSheet'],
+  },
+  styleLayout: {
+    id: '',
+    label: 'Style & Layout',
+    traceEventNames: [
+      'ScheduleStyleRecalculation',
+      'RecalculateStyles',
+      'UpdateLayoutTree',
+      'InvalidateLayout',
+      'Layout',
+      'UpdateLayer',
+      'UpdateLayerTree',
+    ],
+  },
+  paintCompositeRender: {
```

Comment:
I noticed you merged together painting & compositing. I'll survive. :persevere:

Reply:
yeah trying to reduce noise for newbs, if you're digging into the savings/differences there it's time for perf panel/chrome tracing 😛

Reply:
WFM. plus, i think this distinction matters more in the 60fps/jank case than pageload.